Scripts/Runtime/Collections/DeferredNativeArray.cs

```diff
@@ -0,0 +1,333 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Collections
+{
+    [BurstCompatible]
+    [StructLayout(LayoutKind.Sequential)]
+    internal unsafe struct DeferredNativeArrayBufferInfo
+    {
+        public static readonly int SIZE = UnsafeUtility.SizeOf<DeferredNativeArrayBufferInfo>();
+        public static readonly int ALIGNMENT = UnsafeUtility.AlignOf<DeferredNativeArrayBufferInfo>();
+
+        [NativeDisableUnsafePtrRestriction] public void* Buffer;
+        public int Length;
+        public int MaxIndex;
```

Comment:
Nothing. It's used for absolutely nothing. Removed.

I *think* this was a holdover from before when you use indexing on the DeferredArray struct itself like an Array. Which I removed because you might as well just operate on a `NativeArray` directly and then I don't have to re-implement all the same functionality.